dll/win32/browseui/internettoolbar.cpp

```diff
@@ -607,8 +607,24 @@ HRESULT STDMETHODCALLTYPE CMenuCallback::CallbackSM(LPSMDATA psmd, UINT uMsg, WP
 
 CInternetToolbar::CInternetToolbar()
 {
+    HKEY hKey;
+    // Sets the toolbar to locked by default if the registry key is missing.
+    fLocked = true;
+    if(RegOpenKeyEx(HKEY_CURRENT_USER,
+                    _T("Software\\Microsoft\\Internet Explorer\\Toolbar"),
+                    0, KEY_READ, &hKey) == ERROR_SUCCESS)
+    {
+        DWORD dwLocked;
+        DWORD dwSize;
+        if(RegQueryValueEx(hKey,
+                        _T("Locked"),
+                        NULL, NULL, (LPBYTE)&dwLocked, &dwSize) == ERROR_SUCCESS)
+        {
```

Comment:
Maybe check also that the retrieved value is of type `REG_DWORD` and the `dwSize == sizeof(dwLocked)` .

Reply:
Thank you for the suggestion! I added checks to ensure the retrieved registry key is the correct size and type.